### PR TITLE
RelativeTimeStamp localDateMode v2

### PR DESCRIPTION
**Pull request by lbwexler**
This is a refactor of Colin's work in the related branch, that I think better clarifies what we are doing, and how it differs from Moment.

Am curious if folks think we really need the mode 'limited' -- could either discard completely, or fold in to default as Colin has done.

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [X] Caught up with `develop` branch as of last change.
- [X] Added CHANGELOG entry, or determined not required.
- [X] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [X] Updated doc comments / prop-types, or determined not required.
- [X] Reviewed and tested on Mobile, or determined not required.
- [X] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

**Comment by cnrudd** [comment]
I like the 24hr threshhold change
but I don't see why anyone would ever choose localDateMode 'off', or to put it your way: I think we should "fold limited into default [off]".


If the output offered more granular units when the diff spans days, like "1 day, 13hrs", then I could see wanting the choice between `off` and `limited`.

**Comment by lbwexler** [comment]
@cnrudd and I discussed this -- just want to clarify that I think 'limited' is probably correct for most of our applications, and if we really wanted to simplify this API it could be built in.

Also wanted to clarify that I really don't like these names for the modes -- just was at a loss for something better, and wanted to get the PR out there.

File: cmp/relativetimestamp/RelativeTimestamp.ts [hunk 1]
```diff
@@ -62,6 +63,16 @@ export interface RelativeTimestampOptions {
...
     /** Time to which the input timestamp is compared. */
     relativeTo?: Date | number;
+
+    /**
+     * Governs if calendar days should be used for computing string rather than the native
+     * moment 24-hour day. Set to 'limited' to ensure that any output that refers to 'days'
+     * (i.e. between 1 day and 26 days elapsed time) will refer to the difference in
+     * *calendar* days between the timestamps.   Set to 'full' to ensure that any output less
+     * than 25 days will refer to the difference in *calendar* days. Off (default)
+     * will indicate that the standard moment formatting should be used.
```
Comment (TomTirapani) [inline cmp/relativetimestamp/RelativeTimestamp.ts:73]: I find this comment confusing - perhaps it needs more emphasis on the **difference** between full and limited (i.e. "In limited mode, output of less than a day will be left as is")

Comment (lbwexler) [reply cmp/relativetimestamp/RelativeTimestamp.ts:73]: totally agree

**Pull request merged**